File: cacao/ui/components/data/tag/tag.py
"""
Tag component for labeling content with keywords or categories.
"""

from typing import Any, Dict
from ...base import Component


class Tag(Component):
    """
    Labels content with keywords or categories.
    """
    def __init__(
        self,
        content,
        color = None,
        closable = False,
        **kwargs
    ) -> None:
        self.content = content
        self.color = color
        self.closable = closable
        self.extra_props = kwargs

    def render(self) -> Dict[str, Any]:
        return {
            "type": "tag",
            "props": {
                "content": self.content,
                "color": self.color,
                "closable": self.closable,
                **self.extra_props
            }
        }

File: cacao/ui/components/data/tag/tag.js
(component) => {
    console.log("[CacaoCore] Rendering tag component:", component);
    const el = document.createElement("span");
    el.className = "tag";
    
    if (component.props.color) {
        el.classList.add(`tag-${component.props.color}`);
        el.style.backgroundColor = component.props.color;
    }

    if (component.props.content) {
        el.textContent = component.props.content;
    }

    if (component.props.closable) {
        const closeBtn = document.createElement("span");
        closeBtn.className = "tag-close";
        closeBtn.innerHTML = "×";
        closeBtn.onclick = (e) => {
            e.stopPropagation();
            el.remove();
        };
        el.appendChild(closeBtn);
    }

    return el;
}